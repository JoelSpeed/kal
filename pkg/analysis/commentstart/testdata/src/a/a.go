package a

import "a/pkg"

type CommentStartTestStruct struct {
	NoJSONTag     string
	EmptyJSONTag  string `json:""`
	InlineJSONTag string `json:",inline"`
	NoComment     string `json:"noComment"` // want "field NoComment is missing godoc comment"
	Ignored       string `json:"-"`
	Hyphen        string `json:"-,"` // want "field Hyphen is missing godoc comment"

	AnonymousStruct struct { // want "field AnonymousStruct is missing godoc comment"
		NoComment string `json:"noComment"` // want "field NoComment is missing godoc comment"
	} `json:"anonymousStruct"`

	AnonymousStructInlineJSONTag struct {
		NoComment string `json:"noComment"` // want "field NoComment is missing godoc comment"
	} `json:",inline"`

	IgnoredAnonymousStruct struct {
		NoComment string `json:"noComment"`
	} `json:"-"`

	StructForInlineField `json:",inline"`

	A `json:"a"` // want "field A is missing godoc comment"

	PkgA pkg.A `json:"pkgA"` // want "field PkgA is missing godoc comment"

	pkg.Embedded `json:"embedded"` // want "field pkg.Embedded is missing godoc comment"

	*pkg.EmbeddedPointer `json:"embeddedPointer"` // want "field \\*pkg.EmbeddedPointer is missing godoc comment"

	// IncorrectStartComment is a field with an incorrect start to the comment. // want "godoc for field IncorrectStartComment should start with 'incorrectStartComment ...'"
	IncorrectStartComment string `json:"incorrectStartComment"`

	// IncorrectStartOptionalComment is a field with an incorrect start to the comment. // want "godoc for field IncorrectStartOptionalComment should start with 'incorrectStartOptionalComment ...'"
	IncorrectStartOptionalComment string `json:"incorrectStartOptionalComment"`

	// correctStartComment is a field with a correct start to the comment.
	CorrectStartComment string `json:"correctStartComment"`

	// correctStartOptionalComment is a field with a correct start to the comment.
	CorrectStartOptionalComment string `json:"correctStartOptionalComment,omitempty"`

	// IncorrectMultiLineComment is a field with an incorrect start to the comment. // want "godoc for field IncorrectMultiLineComment should start with 'incorrectMultiLineComment ...'"
	// Except this time there are multiple lines to the comment.
	IncorrectMultiLineComment string `json:"incorrectMultiLineComment"`

	// correctMultiLineComment is a field with a correct start to the comment.
	// Except this time there are multiple lines to the comment.
	CorrectMultiLineComment string `json:"correctMultiLineComment"`

	// This comment just isn't correct at all, doesn't even start with anything resembling the field names. // want "godoc for field IncorrectComment should start with 'incorrectComment ...'"
	IncorrectComment string `json:"incorrectComment"`
}

// DoNothing is used to check that the analyser doesn't report on methods.
func (CommentStartTestStruct) DoNothing() {}

type StructForInlineField struct {
	NoComment string `json:"noComment"` // want "field NoComment is missing godoc comment"
}

type A struct {
	NoComment string `json:"noComment"` // want "field NoComment is missing godoc comment"
}

type unexportedStruct struct {
	NoComment string `json:"noComment"` // want "field NoComment is missing godoc comment"
}

type (
	MultipleTypeDeclaration1 struct {
		NoComment string `json:"noComment"` // want "field NoComment is missing godoc comment"
	}
	MultipleTypeDeclaration2 struct {
		NoComment string `json:"noComment"` // want "field NoComment is missing godoc comment"
	}
)

func FunctionWithStructs() {
	type InaccessibleStruct struct {
		NoComment string `json:"noComment"`
	}
}

type Interface interface {
	InaccessibleFunction() string
}
