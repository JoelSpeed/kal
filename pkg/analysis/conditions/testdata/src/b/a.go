package b

import (
	"go/ast"

	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

type ValidConditions struct {
	// conditions is an accurate representation of the desired state of a conditions object.
	// +listType=map
	// +listMapKey=type
	// +patchStrategy=merge
	// +patchMergeKey=type
	// +optional
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`

	// other fields
	OtherField string `json:"otherField,omitempty"`
}

type ConditionsNotFirst struct {
	// other fields
	OtherField string `json:"otherField,omitempty"`

	// conditions is an accurate representation of the desired state of a conditions object.
	// +listType=map
	// +listMapKey=type
	// +patchStrategy=merge
	// +patchMergeKey=type
	// +optional
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,2,rep,name=conditions"`
}

type ConditionsThird struct {
	// other fields
	OtherField string `json:"otherField,omitempty"`

	// another field
	AnotherField string `json:"anotherField,omitempty"`

	// conditions is an accurate representation of the desired state of a conditions object.
	// +listType=map
	// +listMapKey=type
	// +patchStrategy=merge
	// +patchMergeKey=type
	// +optional
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,3,rep,name=conditions"`
}

type ConditionsIncorrectType struct {
	// conditions has an incorrect type.
	Conditions map[string]metav1.Condition // want "Conditions field must be a slice of metav1.Condition"
}

type ConditionsIncorrectSliceElement struct {
	// conditions has an incorrect type.
	Conditions []string // want "Conditions field must be a slice of metav1.Condition"
}

type ConditionsIncorrectImportedSliceElement struct {
	// conditions has an incorrect type.
	Conditions []metav1.Time // want "Conditions field must be a slice of metav1.Condition"
}

type ConditionsIncorrectImportedPackage struct {
	// conditions has an incorrect type.
	Conditions []ast.Node // want "Conditions field must be a slice of metav1.Condition"
}

type MissingAllMarkers struct {
	// conditions is missing all markers.
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"` // want "Conditions field is missing the following markers: listType=map, listMapKey=type, patchStrategy=merge, patchMergeKey=type, optional"
}

type MissingListMarkers struct {
	// conditions is missing list markers.
	// +patchStrategy=merge
	// +patchMergeKey=type
	// +optional
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"` // want "Conditions field is missing the following markers: listType=map, listMapKey=type"
}

type MissingPatchMarkers struct {
	// conditions is missing patch markers.
	// +listType=map
	// +listMapKey=type
	// +optional
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"` // want "Conditions field is missing the following markers: patchStrategy=merge, patchMergeKey=type"
}

type MissingOptionalMarker struct {
	// conditions is missng the optional marker.
	// +listType=map
	// +listMapKey=type
	// +patchStrategy=merge
	// +patchMergeKey=type
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"` // want "Conditions field is missing the following markers: optional"
}

type MissingFieldTag struct {
	// conditions is missing the field tag.
	// +listType=map
	// +listMapKey=type
	// +patchStrategy=merge
	// +patchMergeKey=type
	// +optional
	Conditions []metav1.Condition // want "Conditions field is missing tags, should be: `json:\"conditions,omitempty\" patchStrategy:\"merge\" patchMergeKey:\"type\" protobuf:\"bytes,1,rep,name=conditions\"`"
}

type IncorrectFieldTag struct {
	// conditions has an incorrect field tag.
	// +listType=map
	// +listMapKey=type
	// +patchStrategy=merge
	// +patchMergeKey=type
	// +optional
	Conditions []metav1.Condition `json:"conditions"  patchMergeKey:"type" protobuf:"bytes,3,rep,name=conditions"` // want "Conditions field has incorrect tags, should be: `json:\"conditions,omitempty\" patchStrategy:\"merge\" patchMergeKey:\"type\" protobuf:\"bytes,1,rep,name=conditions\"`"
}
