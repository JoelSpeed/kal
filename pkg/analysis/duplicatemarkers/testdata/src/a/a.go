package a

// +kubebuilder:validation:Enum=foo;bar;baz
// +optional
// +kubebuilder:validation:Enum=foo;bar;baz
type Enum string // want "Enum has duplicated markers kubebuilder:validation:Enum"

// It must be ignored since it is not a type
// +required
var Variable string

// +kubebuilder:object:root=true
// +kubebuilder:subresource:status
// +kubebuilder:object:root=true
type DuplicateMarkerSpec struct { // want "DuplicateMarkerSpec has duplicated markers kubebuilder:object:root"
	// +kubebuilder:validation:Required
	Foo string `json:"foo"`

	// +kubebuilder:validation:Required
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MaxLength=10
	DuplicatedFoo string `json:"duplicatedFoo"` // want "DuplicatedFoo has duplicated markers kubebuilder:validation:Required"
}
