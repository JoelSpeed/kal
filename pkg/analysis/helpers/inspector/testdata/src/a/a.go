package A

var (
	String string
)

const (
	Int int = 0
)

type A struct {
	Field string `json:"field"` // want "field: Field" "json tag: field"

	B `json:"b"` // want "field: B" "json tag: b"

	C `json:",inline"` // want "field: C"

	D `json:"-"`

	E struct { // want "field: E" "json tag: e"
		Field string `json:"field"` // want "field: Field" "json tag: field"
	} `json:"e"`

	F struct {
		Field string `json:"field"`
	} `json:"-"`
}

func (A) DoNothing() {}

type B struct {
	Field string `json:"field"` // want "field: Field" "json tag: field"
}

type (
	C struct {
		Field string `json:"field"` // want "field: Field" "json tag: field"
	}

	D struct {
		Field string `json:"field"` // want "field: Field" "json tag: field"
	}
)

func Foo() {
	type Bar struct {
		Field string
	}
}

type Bar interface {
	Name() string
}

var Var = struct {
	Field string
}{
	Field: "field",
}
