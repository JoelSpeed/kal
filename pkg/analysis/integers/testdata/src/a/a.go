package a

type Integers struct {
	ValidString string

	ValidMap map[string]string

	ValidInt32 int32

	ValidInt32Ptr *int32

	ValidInt64 int64

	ValidInt64Ptr *int64

	InvalidInt int // want "field InvalidInt should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidIntPtr *int // want "field InvalidIntPtr pointer should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidInt8 int8 // want "field InvalidInt8 should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidInt16 int16 // want "field InvalidInt16 should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidUInt uint // want "field InvalidUInt should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidUIntPtr uint // want "field InvalidUIntPtr should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidUInt8 uint8 // want "field InvalidUInt8 should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidUInt16 uint16 // want "field InvalidUInt16 should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidUInt32 uint32 // want "field InvalidUInt32 should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidUInt64 uint64 // want "field InvalidUInt64 should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	ValidInt32Alias ValidInt32Alias

	ValidInt32AliasPtr *ValidInt32Alias

	InvalidIntAlias InvalidIntAlias // want "field InvalidIntAlias type InvalidIntAlias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidIntAliasPtr *InvalidIntAlias // want "field InvalidIntAliasPtr pointer type InvalidIntAlias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidUIntAlias InvalidUIntAlias // want "field InvalidUIntAlias type InvalidUIntAlias should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidUIntAliasPtr *InvalidUIntAlias // want "field InvalidUIntAliasPtr pointer type InvalidUIntAlias should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidIntAliasAlias InvalidIntAliasAlias // want "field InvalidIntAliasAlias type InvalidIntAliasAlias type InvalidIntAlias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	ValidSliceInt32 []int32

	ValidSliceInt32Ptr []*int32

	ValidSliceInt64 []int64

	ValidSliceInt64Ptr []*int64

	InvalidSliceInt []int // want "field InvalidSliceInt array element should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidSliceIntPtr []*int // want "field InvalidSliceIntPtr array element pointer should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidSliceUInt []uint // want "field InvalidSliceUInt array element should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidSliceUIntPtr []*uint // want "field InvalidSliceUIntPtr array element pointer should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidSliceIntAlias []InvalidIntAlias // want "field InvalidSliceIntAlias array element type InvalidIntAlias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidSliceIntAliasPtr []*InvalidIntAlias // want "field InvalidSliceIntAliasPtr array element pointer type InvalidIntAlias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidSliceUIntAlias []InvalidUIntAlias // want "field InvalidSliceUIntAlias array element type InvalidUIntAlias should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidSliceUIntAliasPtr []*InvalidUIntAlias // want "field InvalidSliceUIntAliasPtr array element pointer type InvalidUIntAlias should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	ValidMapStringToInt32 map[string]int32

	ValidMapStringToInt64 map[string]int64

	InvalidMapStringToInt map[string]int // want "field InvalidMapStringToInt map value should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidMapStringToUInt map[string]uint // want "field InvalidMapStringToUInt map value should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	ValidMapInt32ToString map[int32]string

	ValidMapInt64ToString map[int64]string

	InvalidMapIntToString map[int]string // want "field InvalidMapIntToString map key should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidMapUIntToString map[uint]string // want "field InvalidMapUIntToString map key should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

	InvalidIntFromAnotherFile IntB // want "field InvalidIntFromAnotherFile type IntB should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

	InvalidSliceIntAliasFromAnotherFile InvalidSliceIntAliasB // want "field InvalidSliceIntAliasFromAnotherFile type InvalidSliceIntAliasB array element should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"
}

// DoNothing is used to check that the analyser doesn't report on methods.
func (Integers) DoNothing(a int, b uint) (int, uint) {
	return 0, 0
}

type ValidInt32Alias int32

type ValidInt32PtrAlias *int32

type ValidInt64Alias int64

type ValidInt64PtrAlias *int64

type InvalidIntAlias int // want "type InvalidIntAlias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

type InvalidIntPtrAlias *int // want "type InvalidIntPtrAlias pointer should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

type InvalidInt8Alias int8 // want "type InvalidInt8Alias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

type InvalidInt16Alias int16 // want "type InvalidInt16Alias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

type InvalidUIntAlias uint // want "type InvalidUIntAlias should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

type InvalidUIntPtrAlias *uint // want "type InvalidUIntPtrAlias pointer should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

type InvalidUInt8Alias uint8 // want "type InvalidUInt8Alias should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

type InvalidUInt16Alias uint16 // want "type InvalidUInt16Alias should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

type InvalidUInt32Alias uint32 // want "type InvalidUInt32Alias should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

type InvalidUInt64Alias uint64 // want "type InvalidUInt64Alias should not use unsigned integers, use only int32 or int64 and apply validation to ensure the value is positive"

type InvalidIntAliasAlias InvalidIntAlias // want "type InvalidIntAliasAlias type InvalidIntAlias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

type InvalidSliceIntAlias []int // want "type InvalidSliceIntAlias array element should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

type InvalidSliceIntPtrAlias []*int // want "type InvalidSliceIntPtrAlias array element pointer should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

type InvalidSliceIntAliasAlias []InvalidIntAlias // want "type InvalidSliceIntAliasAlias array element type InvalidIntAlias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"

type InvalidSliceIntAliasPtrAlias []*InvalidIntAlias // want "type InvalidSliceIntAliasPtrAlias array element pointer type InvalidIntAlias should not use an int, int8 or int16. Use int32 or int64 depending on bounding requirements"
