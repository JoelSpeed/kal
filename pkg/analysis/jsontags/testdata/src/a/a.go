package a

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

type JSONTagTestStruct struct {
	NoJSONTag            string // want "field NoJSONTag is missing json tag"
	EmptyJSONTag         string `json:""`                        // want "field EmptyJSONTag has empty json tag"
	NonCamelCaseJSONTag  string `json:"non_camel_case_json_tag"` // want "field NonCamelCaseJSONTag json tag does not match pattern \"\\^\\[a-z\\]\\[a-z0-9\\]\\*\\(\\?:\\[A-Z\\]\\[a-z0-9\\]\\*\\)\\*\\$\": non_camel_case_json_tag"
	WithHyphensJSONTag   string `json:"with-hyphens-json-tag"`   // want "field WithHyphensJSONTag json tag does not match pattern \"\\^\\[a-z\\]\\[a-z0-9\\]\\*\\(\\?:\\[A-Z\\]\\[a-z0-9\\]\\*\\)\\*\\$\": with-hyphens-json-tag"
	PascalCaseJSONTag    string `json:"PascalCaseJSONTag"`       // want "field PascalCaseJSONTag json tag does not match pattern \"\\^\\[a-z\\]\\[a-z0-9\\]\\*\\(\\?:\\[A-Z\\]\\[a-z0-9\\]\\*\\)\\*\\$\": PascalCaseJSONTag"
	NonTerminatedJSONTag string `json:"nonTerminatedJSONTag`     // want "field NonTerminatedJSONTag is missing json tag"
	XMLTag               string `xml:"xmlTag"`                   // want "field XMLTag is missing json tag"
	InlineJSONTag        string `json:",inline"`
	ValidJSONTag         string `json:"validJsonTag"`
	ValidOptionalJSONTag string `json:"validOptionalJsonTag,omitempty"`
	JSONTagWithID        string `json:"jsonTagWithID"`
	JSONTagWithTTL       string `json:"jsonTagWithTTL"`
	JSONTagWithGiB       string `json:"jsonTagWithGiB"`

	A `json:",inline"`
	B `json:"bar,omitempty"`
	C             // want "embedded field C is missing json tag"
	D `json:""`   // want "embedded field D has empty json tag"
	E `json:"e-"` // want "embedded field E json tag does not match pattern \"\\^\\[a-z\\]\\[a-z0-9\\]\\*\\(\\?:\\[A-Z\\]\\[a-z0-9\\]\\*\\)\\*\\$\": e-"

	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`
}

type A struct{}

func (A) DoNothing() {}

type B struct{}

type C struct{}

type D struct{}

type E struct{}
