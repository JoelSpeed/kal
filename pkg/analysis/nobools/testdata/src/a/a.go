package a

type Integers struct {
	ValidString string

	ValidMap map[string]string

	ValidInt32 int32

	ValidInt64 int64

	InvalidBool bool // want "field InvalidBool should not use a bool. Use a string type with meaningful constant values as an enum."

	InvalidBoolPtr *bool // want "field InvalidBoolPtr pointer should not use a bool. Use a string type with meaningful constant values as an enum."

	InvalidBoolSlice []bool // want "field InvalidBoolSlice array element should not use a bool. Use a string type with meaningful constant values as an enum."

	InvalidBoolPtrSlice []*bool // want "field InvalidBoolPtrSlice array element pointer should not use a bool. Use a string type with meaningful constant values as an enum."

	InvalidBoolAlias BoolAlias // want "field InvalidBoolAlias type BoolAlias should not use a bool. Use a string type with meaningful constant values as an enum."

	InvalidBoolPtrAlias *BoolAlias // want "field InvalidBoolPtrAlias pointer type BoolAlias should not use a bool. Use a string type with meaningful constant values as an enum."

	InvalidBoolSliceAlias []BoolAlias // want "field InvalidBoolSliceAlias array element type BoolAlias should not use a bool. Use a string type with meaningful constant values as an enum."

	InvalidBoolPtrSliceAlias []*BoolAlias // want "field InvalidBoolPtrSliceAlias array element pointer type BoolAlias should not use a bool. Use a string type with meaningful constant values as an enum."
}

type BoolAlias bool // want "type BoolAlias should not use a bool. Use a string type with meaningful constant values as an enum."

type BoolAliasPtr *bool // want "type BoolAliasPtr pointer should not use a bool. Use a string type with meaningful constant values as an enum."

type BoolAliasSlice []bool // want "type BoolAliasSlice array element should not use a bool. Use a string type with meaningful constant values as an enum."

type BoolAliasPtrSlice []*bool // want "type BoolAliasPtrSlice array element pointer should not use a bool. Use a string type with meaningful constant values as an enum."
