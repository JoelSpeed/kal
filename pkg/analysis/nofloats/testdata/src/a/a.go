package a

type Floats struct {
	ValidString string

	ValidMap map[string]string

	ValidInt32 int32

	ValidInt64 int64

	InvalidFloat32 float32 // want "field InvalidFloat32 should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat64 float64 // want "field InvalidFloat64 should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat32Ptr *float32 // want "field InvalidFloat32Ptr pointer should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat64Ptr *float64 // want "field InvalidFloat64Ptr pointer should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat32Slice []float32 // want "field InvalidFloat32Slice array element should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat64Slice []float64 // want "field InvalidFloat64Slice array element should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat32PtrSlice []*float32 // want "field InvalidFloat32PtrSlice array element pointer should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat64PtrSlice []*float64 // want "field InvalidFloat64PtrSlice array element pointer should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat32Alias Float32Alias // want "field InvalidFloat32Alias type Float32Alias should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat64Alias Float64Alias // want "field InvalidFloat64Alias type Float64Alias should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat32PtrAlias *Float32Alias // want "field InvalidFloat32PtrAlias pointer type Float32Alias should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat64PtrAlias *Float64Alias // want "field InvalidFloat64PtrAlias pointer type Float64Alias should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat32SliceAlias []Float32Alias // want "field InvalidFloat32SliceAlias array element type Float32Alias should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat64SliceAlias []Float64Alias // want "field InvalidFloat64SliceAlias array element type Float64Alias should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat32PtrSliceAlias []*Float32Alias // want "field InvalidFloat32PtrSliceAlias array element pointer type Float32Alias should not use a float value because they cannot be reliably round-tripped."

	InvalidFloat64PtrSliceAlias []*Float64Alias // want "field InvalidFloat64PtrSliceAlias array element pointer type Float64Alias should not use a float value because they cannot be reliably round-tripped."

	InvalidMapStringToFloat32 map[string]float32 // want "field InvalidMapStringToFloat32 map value should not use a float value because they cannot be reliably round-tripped."

	InvalidMapStringToFloat64 map[string]float64 // want "field InvalidMapStringToFloat64 map value should not use a float value because they cannot be reliably round-tripped."

	InvalidMapStringToFloat32Ptr map[string]*float32 // want "field InvalidMapStringToFloat32Ptr map value pointer should not use a float value because they cannot be reliably round-tripped."

	InvalidMapStringToFloat64Ptr map[string]*float64 // want "field InvalidMapStringToFloat64Ptr map value pointer should not use a float value because they cannot be reliably round-tripped."

	InvalidMapFloat32ToString map[float32]string // want "field InvalidMapFloat32ToString map key should not use a float value because they cannot be reliably round-tripped."

	InvalidMapFloat64ToString map[float64]string // want "field InvalidMapFloat64ToString map key should not use a float value because they cannot be reliably round-tripped."

	InvalidMapFloat32PtrToString map[*float32]string // want "field InvalidMapFloat32PtrToString map key pointer should not use a float value because they cannot be reliably round-tripped."

	InvalidMapFloat64PtrToString map[*float64]string // want "field InvalidMapFloat64PtrToString map key pointer should not use a float value because they cannot be reliably round-tripped."
}

// DoNothingFloat32 is used to check that the analyser doesn't report on methods.
func (Floats) DoNothingFloat32(a float32) float32 {
	return a
}

// DoNothingFloat64 is used to check that the analyser doesn't report on methods.
func (Floats) DoNothingFloat64(a float64) float64 {
	return a
}

type Float32Alias float32 // want "type Float32Alias should not use a float value because they cannot be reliably round-tripped."

type Float64Alias float64 // want "type Float64Alias should not use a float value because they cannot be reliably round-tripped."

type Float32AliasPtr *float32 // want "type Float32AliasPtr pointer should not use a float value because they cannot be reliably round-tripped."

type Float64AliasPtr *float64 // want "type Float64AliasPtr pointer should not use a float value because they cannot be reliably round-tripped."

type Float32AliasSlice []float32 // want "type Float32AliasSlice array element should not use a float value because they cannot be reliably round-tripped."

type Float64AliasSlice []float64 // want "type Float64AliasSlice array element should not use a float value because they cannot be reliably round-tripped."

type Float32AliasPtrSlice []*float32 // want "type Float32AliasPtrSlice array element pointer should not use a float value because they cannot be reliably round-tripped."

type Float64AliasPtrSlice []*float64 // want "type Float64AliasPtrSlice array element pointer should not use a float value because they cannot be reliably round-tripped."

type MapStringToFloat32Alias map[string]float32 // want "type MapStringToFloat32Alias map value should not use a float value because they cannot be reliably round-tripped."

type MapStringToFloat64Alias map[string]float64 // want "type MapStringToFloat64Alias map value should not use a float value because they cannot be reliably round-tripped."

type MapStringToFloat32PtrAlias map[string]*float32 // want "type MapStringToFloat32PtrAlias map value pointer should not use a float value because they cannot be reliably round-tripped."

type MapStringToFloat64PtrAlias map[string]*float64 // want "type MapStringToFloat64PtrAlias map value pointer should not use a float value because they cannot be reliably round-tripped."
