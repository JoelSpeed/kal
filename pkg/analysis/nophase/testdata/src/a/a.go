package a

type NoPhaseTestStruct struct {
	// +optional
	Phase *string `json:"phase,omitempty"` // want "field Phase: phase fields are deprecated and conditions should be preferred, avoid phase like enum fields"

}

type NoSubPhaseTestStruct struct {
	// +optional
	FooPhase *string `json:"fooPhase,omitempty"` // want "field FooPhase: phase fields are deprecated and conditions should be preferred, avoid phase like enum fields"

}
