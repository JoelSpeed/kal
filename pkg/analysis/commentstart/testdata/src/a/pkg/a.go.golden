package pkg

type A struct {
	NoComment string `json:"noComment"` // want "field NoComment is missing godoc comment"
}

// To embed the same struct multiple times, we need to rename the type.
type Embedded A
type EmbeddedPointer A
